platform/api/gcloud/compute.go

```diff
@@ -46,10 +46,17 @@ func (a *API) mkinstance(userdata, name string, keys []*agent.Key) *compute.Inst
 		})
 	}
 
-	prefix := "https://www.googleapis.com/compute/v1/projects/" + a.options.Project
+	instancePrefix := "https://www.googleapis.com/compute/v1/projects/" + a.options.Project
+
+	// If long image name given for image, respect the image project
+	imagePrefix := instancePrefix
+	if strings.HasPrefix(a.options.Image, "https") {
```

Comment:
I don't know about other tools off the top of my head but the old gcutil accepted image names relative to the API endpoint, e.g. `projects/name/global/images/foo`

So how about this to be as flexible as possible:
 - If prefix is the API endpoint `https://www.googleapis.com/compute/v1/` pass the value through verbatim.
 - If prefix is `projects/` append it to the URL endpoint and use it.
 - If image contains no `/` construct the appropriate URL using the default project.
 - Otherwise the name is invalid.